crates/autopilot/src/run_loop.rs

```diff
@@ -398,16 +391,20 @@ impl RunLoop {
         }
     }
 
-    #[allow(clippy::too_many_arguments)]
     async fn post_processing(
         &self,
         auction: &domain::Auction,
         competition_simulation_block: u64,
-        winning_solution: &competition::Solution,
-        solutions: &[Participant],
+        solutions: &[competition::Participant],
         block_deadline: u64,
     ) -> Result<()> {
         let start = Instant::now();
+        // TODO: Support multiple winners
+        // https://github.com/cowprotocol/services/issues/3021
+        let Some(winning_solution) = solutions.first().map(|participant| &participant.solution)
```

Comment:
This is now a bit awkward because the type system doesn't prevent a non-winning order to come first. 🤔

Reply:
Impossible for the same reason as in https://github.com/cowprotocol/services/pull/3042#discussion_r1793162627

Reply:
Added finding of winner for the piece of mind, afterall.